controllers/activemqartemis_controller.go

```diff
@@ -405,6 +412,36 @@ func (r *ActiveMQArtemisReconcilerImpl) validateExposeModes(customResource *brok
 	return nil, false
 }
 
+func (r *ActiveMQArtemisReconcilerImpl) validateEnvVars(customResource *brokerv1beta1.ActiveMQArtemis) (*metav1.Condition, bool) {
+
+	internalVarNames := map[string]string{
+		debugArgsEnvVarName:      debugArgsEnvVarName,
+		javaOptsEnvVarName:       javaOptsEnvVarName,
+		javaArgsAppendEnvVarName: javaArgsAppendEnvVarName,
+	}
+
+	invalidVars := []string{}
+
+	for _, envVar := range customResource.Spec.Env {
+		if _, ok := internalVarNames[envVar.Name]; ok {
+			if envVar.ValueFrom != nil {
+				invalidVars = append(invalidVars, envVar.Name)
+			}
+		}
+	}
+
+	if len(invalidVars) > 0 {
+		return &metav1.Condition{
+			Type:    brokerv1beta1.ValidConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  brokerv1beta1.ValidConditionInvalidInternalVarUsage,
+			Message: fmt.Sprintf("Using valueFrom with internal env vars %v will get ignored. Instead use a different var and refernece it in its value field.", invalidVars),
```

Comment:
Maybe just. Don't use valueFrom on env vars that the operator can mutate.

... and reference as a dependent var....

Reply:
The ignored but seems wrong and the kube docs call out dependent vars

Reply:
Updated.